### DISCUSSION TITLE
Current project status

### DISCUSSION THREAD

**Discussion by golf1052** [signal-csharp/Signal-Windows#217] [General]
Happy 2021! Wanted to give an update on the current project status. As with any open source project the project is alive as long as people make contributions. Pull requests are always welcome.

The official [Signal Android app](https://github.com/signalapp/Signal-Android), our reference implementation, has at least 4 full time employees working on it with hundreds of pull requests from the community. This project has a few people working on it in their spare time with not much community support. This isn't to discourage but it does mean that updates will be much slower than the official Signal project.

I've updated curve25519-dotnet, libsignal-protocol-dotnet, and libsignal-metadata-dotnet to the latest official Signal implementations and now work continues on updating libsignal-service-dotnet (which is at least 250 commits behind libsignal-service-java), making bug fixes, and adding new features in this app.

**Comment by Trolldemorted** [comment]
Do you think we can implement the new groups soonish? I fear that signal-windows users can't be added to new-style groups.

**Reply by golf1052** [reply]
I'll try to make that my next priority. After fixing https://github.com/signal-csharp/Signal-Windows/issues/212 (hopefully implementing the contact discovery service fixes that) the other things I want to work on would be fixing attachments on mobile or adding dark mode.

**Comment by KhaaL** [comment]
I'd be happy to contribute with bug-reporting and writing documentation 😀 Is there any onboarding document for new recruits?

**Reply by golf1052** [reply]
Right now the only on-boarding documentation we have is https://github.com/signal-csharp/Signal-Windows/blob/master/CONTRIBUTING.md

It would be nice to have an architecture document that goes over the high level architecture (important classes and how they interact). It would be as described in [this blog post](https://matklad.github.io//2021/02/06/ARCHITECTURE.md.html) with [this (rust-analyzer)](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/architecture.md) and [this (caddy)](https://caddyserver.com/docs/architecture) as examples.

**Reply by KhaaL** [reply]
> It would be nice to have an architecture document that goes over the high level architecture 

I would *love* to do that, but I'm afraid I still lack the competence and in-depth technical understanding to write such a document. I'll keep an eye out on the issues here and see if anything arise that I can contribute with 😃
